src/plugins/kibana/server/lib/case_conversion.js

```diff
@@ -0,0 +1,15 @@
+const _ = require('lodash');
+
+module.exports = {
+  convertToSnakeCase: function (object) {
+    return _.mapKeys(object, (value, key) => {
+      return _.snakeCase(key);
+    });
+  },
+
+  convertToCamelCase: function (object) {
```

Comment:
If we want to follow the path of libraries like lodash, helpers should be shallow by default and be suffixed with `Deep` in the event that they are recursive.